src/ifcbondcollector.py

```diff
@@ -101,69 +103,79 @@ def get_link_stat(self, ifc_list):
 
         # collecting data to publish
         for ifc_name in ifc_list.keys():
-            self.publish_data["ifc"].setdefault(ifc_name,{})
-            self.publish_data["ifc"][ifc_name]['Status'] = (True if ifc_list.get(ifc_name).get('MII Status') == 'up' else False)  
+            self.publish_data["ifc"].setdefault(ifc_name, {})
+            self.publish_data["ifc"][ifc_name]['Status'] = ifc_list.get(ifc_name).get('MII Status') == 'up'
             self.publish_data["ifc"][ifc_name]['Link Failure Count'] = ifc_list.get(ifc_name).get('Link Failure Count')
-            self.publish_data["ifc"][ifc_name]['Active'] =  ifc_list.get(ifc_name).get('active')
-            self.publish_data["ifc"][ifc_name]['lldp_stats_unavailable'] = (True if len(ifc_list.get(ifc_name).get('lldp_stats')) <1 else False)
+            self.publish_data["ifc"][ifc_name]['Active'] = ifc_list.get(ifc_name).get('active')
+            self.publish_data["ifc"][ifc_name]['lldp_stats_unavailable'] = len(ifc_list.get(ifc_name).get('lldp_stats')) < 1
 
         return (ifc_list)
 
     def check_given_rules(self, ifc_list, rules):
         """ compare parameters from ifcbondingcollector.conf with parsed data in lldpctl """
 
         if rules:
-            self.publish_data.setdefault('rules',{})
+            self.publish_data.setdefault('rules', {})
             for rule_name in rules:
                 ifc_name = self.config.get(rule_name)[0]
-                check_parametr = self.config.get(rule_name)[1]
-                value_parametr = self.config.get(rule_name)[2]
-
-                check_rule = ( True if ifc_name not in ifc_list.keys() or value_parametr != ifc_list.get(ifc_name,{}).get('lldp_stats',{}).get(check_parametr,{}) else False )
-                self.publish_data['rules'][rule_name] = check_rule
+                chk_param = self.config.get(rule_name)[1]
+                val_param = self.config.get(rule_name)[2]
 
+                val_param_match = val_param == ifc_list.get(ifc_name, {}).get('lldp_stats', {}).get(chk_param, {})
+                chk_rule = ifc_name not in ifc_list.keys() or not val_param_match
+                self.publish_data['rules'][rule_name] = chk_rule
 
     def check_bonding_match(self, ifc_list):
         """ validation connection to switches """
 
         dev_list = []
-        self.publish_data.setdefault('mismatch_bond',{})
+        self.publish_data.setdefault('mismatch_bond', {})
 
         for ifc_name in ifc_list.keys():
-            dev_list.append(ifc_list.get(ifc_name,{}).get('lldp_stats',{}).get('chassis_name',{}))
+            dev_list.append(ifc_list.get(ifc_name, {}).get('lldp_stats', {}).get('chassis_name', {}))
 
-        self.publish_data['mismatch_bond'] = ( dev_list[1:] == dev_list[:-1] )
+        self.publish_data['mismatch_bond'] = dev_list[1:] == dev_list[:-1]
 
+    def _publish(self, name, value):
+        self.publish(name, value)
+        logging.debug('%s %s', name, value)
```

Comment:
It would be nice if it was more descriptive

Reply:
You're right. Done